### ISSUE TITLE
Flake unit test

### ISSUE THREAD

**Issue by zhang2014** [datafuselabs/datafuse#1261]
**Summary**
```
make test 
bash ./scripts/ci/ci-run-unit-tests.sh
Starting unit tests
warning: Both `/Users/coswde/.cargo/config` and `/Users/coswde/.cargo/config.toml` exist. Using `/Users/coswde/.cargo/config`
warning: Both `/Users/coswde/.cargo/config` and `/Users/coswde/.cargo/config.toml` exist. Using `/Users/coswde/.cargo/config`
   Compiling common-flights v0.1.0 (/Users/coswde/Source/fuse-query/common/flights)
   Compiling fuse-query v0.1.0 (/Users/coswde/Source/fuse-query/fusequery/query)
   Compiling fuse-store v0.1.0 (/Users/coswde/Source/fuse-query/fusestore/store)
   Compiling fuzz v0.1.0 (/Users/coswde/Source/fuse-query/fusequery/fuzz)
    Finished test [unoptimized + debuginfo] target(s) in 1m 12s
     Running unittests (target/debug/deps/common_allocators-4d08b180fc9b1871)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_arrow-b0bda7669bbe9ac9)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_building-7eb089a239532c0d)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_datablocks-399769a71f258da9)

running 8 tests
test kernels::data_block_group_by_hash_test::test_data_block_group_by_hash ... ok
test kernels::data_block_slice_test::test_data_block_slice ... ok
test kernels::data_block_take_test::test_data_block_take ... ok
test kernels::data_block_scatter_test::test_data_block_scatter ... ok
test kernels::data_block_sort_test::test_data_block_sort ... ok
test kernels::data_block_concat_test::test_data_block_concat ... ok
test kernels::data_block_group_by_test::test_data_block_group_by ... ok
test data_block_test::test_data_block ... ok

test result: ok. 8 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.36s

     Running unittests (target/debug/deps/common_datavalues-37f3a477bbcb3ab5)

running 20 tests
test arrays::data_array_test::test_data_array ... ok
test arrays::ops::agg_test::test_boolean_array_agg ... ok
test arrays::ops::apply_test::test_boolean_array_apply ... ok
test arrays::ops::agg_test::test_array_agg ... ok
test arrays::ops::apply_test::test_array_apply ... ok
test arrays::ops::apply_test::test_array_apply_kernel ... ok
test arrays::ops::agg_test::test_utf8_array_agg ... ok
test arrays::kernels::take::test::test_utf8_kernel ... ok
test arrays::ops::apply_test::test_boolean_array_apply_kernel ... ok
test arrays::ops::apply_test::test_utf8_array_apply ... ok
test arrays::ops::cast_test::test_array_cast ... ok
test arrays::ops::group_hash_test::test_group_hash ... ok
test arrays::ops::downcast_test::test_array_as_ref ... ok
test arrays::ops::fill_test::test_array_fill ... ok
test vec_test::test_aligned_vec_allocations ... ok
test arrays::ops::downcast_test::test_array_downcast ... ok
test series::arithmetic_test::test_arithmetic_simple_series ... ok
test data_array_filter_test::filter_batch_array ... ok
test arrays::ops::scatter_test::test_scatter ... ok
test series::arithmetic_test::test_arithmetic_series ... ok

test result: ok. 20 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.67s

     Running unittests (target/debug/deps/common_exception-9528d15b88f0222b)

running 4 tests
test exception_test::test_derive_from_display ... ok
test exception_test::test_from_and_to_status ... ok
test exception_test::test_derive_from_std_error ... ok
test exception_test::test_format_with_error_codes ... ok

test result: ok. 4 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.31s

     Running tests/prelude_test.rs (target/debug/deps/prelude_test-076d49a0f13403df)

running 1 test
test test_prelude ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.28s

     Running unittests (target/debug/deps/common_flights-cd15e3268d683554)

running 2 tests
test impls::storage_api_impl_utils_test::test::test_get_set_meta ... ok
test dns_resolver_test::test_resolver_github ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.15s

     Running unittests (target/debug/deps/common_functions-4712721363466a21)

running 16 tests
test scalars::arithmetics::arithmetic_test::test_arithmetic_function ... ok
test scalars::function_column_test::test_column_function ... ok
test scalars::expressions::cast_test::test_cast_function ... ok
test aggregates::aggregate_combinator_test::test_aggregate_combinator_function ... ok
test scalars::logics::logic_test::test_logic_function ... ok
test scalars::hashes::siphash_test::test_siphash_function ... ok
test scalars::udfs::database_test::test_database_function ... ok
test scalars::udfs::to_type_name_test::test_to_type_name_function ... ok
test scalars::udfs::udf_example_test::test_udf_example_function ... ok
test scalars::strings::substring_test::test_substring_function ... ok
test scalars::udfs::version_test::test_version_function ... ok
test scalars::comparisons::comparison_test::test_comparison_function ... ok
test aggregates::aggregate_batch_test::test_aggregate_batch ... ok
test aggregates::aggregate_function_test::test_aggregate_function ... ok
test aggregates::aggregate_combinator_test::test_aggregate_combinator_function_on_empty_data ... ok
test aggregates::aggregate_function_test::test_aggregate_function_on_empty_data ... ok

test result: ok. 16 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.16s

     Running unittests (target/debug/deps/common_infallible-d869761921f2d645)

running 2 tests
test mutex_test::test_mutex ... ok
test rwlock_test::test_rwlock ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_io-85719ba4a8b2b2c5)

running 13 tests
test binary_write_test::test_put_uvarint ... ok
test binary_read_test::test_write_and_read ... ok
test marshal_test::test_i32 ... ok
test marshal_test::test_bool ... ok
test marshal_test::test_f32 ... ok
test marshal_test::test_i16 ... ok
test marshal_test::test_f64 ... ok
test marshal_test::test_i64 ... ok
test marshal_test::test_i8 ... ok
test marshal_test::test_u16 ... ok
test marshal_test::test_u32 ... ok
test marshal_test::test_u8 ... ok
test marshal_test::test_u64 ... ok

test result: ok. 13 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_management-1ccbc7a04481f1eb)

running 18 tests
test user::user_mgr_test::drop::test_drop_user_normal_case ... ok
test user::user_mgr_test::get::test_get_user_do_not_care_seq ... ok
test user::user_mgr_test::get::test_get_user_seq_match ... ok
test user::user_mgr_test::get_all_users::test_get_all_users_normal ... ok
test user::user_mgr_test::get_users::test_get_users_normal ... ok
test user::user_mgr_test::test_user_info_converter ... ok
test user::user_mgr_test::get::test_get_user_invalid_user_info_encoding ... ok
test user::user_mgr_test::update::test_update_user_none_update ... ok
test user::user_mgr_test::update::test_update_user_normal_full_update ... ok
test user::user_mgr_test::update::test_update_user_normal_partial_update ... ok
test user::user_mgr_test::drop::test_drop_user_unknown ... ok
test user::user_mgr_test::update::test_update_user_full_unknown ... ok
test user::user_mgr_test::get_all_users::test_get_all_users_invalid_user_info_encoding ... ok
test user::user_mgr_test::update::test_update_user_partial_unknown ... ok
test user::user_mgr_test::get::test_get_user_not_exist ... ok
test user::user_mgr_test::get_users::test_get_users_invalid_user_info_encoding ... ok
test user::user_mgr_test::get::test_get_user_not_exist_seq_mismatch ... ok
test user::user_mgr_test::add::test_add_user ... ok

test result: ok. 18 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.57s

     Running unittests (target/debug/deps/common_metatypes-8938eaae2400fadd)

running 3 tests
test match_seq_test::test_match_seq_from_opt_u64 ... ok
test match_seq_test::test_match_seq_match_seq_value ... ok
test match_seq_test::test_match_seq_display ... ok

test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_planners-835073005f7620d5)

running 18 tests
test plan_describe_table_test::test_describe_table_plan ... ok
test plan_expression_test::test_expression_plan_format ... ok
test plan_extras_test::test_plan_extras ... ok
test plan_explain_test::test_explain_plan ... ok
test plan_display_test::test_plan_display_indent ... ok
test plan_expression_test::test_expression_plan ... ok
test plan_filter_test::test_filter_plan ... ok
test plan_limit_test::test_limit_plan ... ok
test plan_having_test::test_having_plan ... ok
test plan_projection_test::test_projection_plan ... ok
test plan_aggregator_test::test_aggregator_plan ... ok
test plan_scan_test::test_scan_plan ... ok
test plan_rewriter_test::test_rewrite_expressions_plan ... ok
test plan_select_test::test_select_wildcard_plan ... ok
test plan_rewriter_test::test_plan_rewriter_1 ... ok
test plan_builder_test::test_plan_builds ... ok
test plan_rewriter_test::test_rewrite_projection_alias_plan ... ok
test plan_expression_test::test_expression_validate ... ok

test result: ok. 18 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.97s

     Running unittests (target/debug/deps/common_profling-3c391e715e97c8fd)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_progress-24334366e6ec91a0)

running 1 test
test progress_test::test_progress ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_runtime-8c572bef0d0688bb)

running 1 test
test runtime_test::test_runtime ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_stoppable-131eae1d690c056c)

running 3 tests
test stoppable_test::test_stop_handle_drop ... ok
test stoppable_test::test_stoppable ... ok
test stoppable_test::test_stop_handle ... ok

test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.87s

     Running unittests (target/debug/deps/common_store_api-e460744415a4bfdb)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_streams-f5f608fbacc13cba)

running 2 tests
test stream_datablock_test::test_datablock_stream ... ok
test stream_progress_test::test_progress_stream ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/common_tracing-39da6e238fd73515)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/datafuse_cli-6fc0f9dacc7b7d48)

running 1 test
test cmds::status_test::test_status ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/datafuse_cli-62384eed2251a672)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/fuse_query-025035e301cfdf41)

running 99 tests
test api::http::v1::cluster_test::test_cluster ... ok
test api::rpc::flight_actions_test::test_shuffle_action_try_into ... ok
test api::rpc::flight_dispatcher_test::test_run_shuffle_action_with_no_scatters ... ok
test api::rpc::flight_tickets_test::test_stream_ticket_try_into ... ok
test api::rpc::flight_service_test::test_do_flight_action_with_different_session ... ok
test clusters::address_test::test_serialize_address ... ok
test api::rpc::flight_dispatcher_test::test_run_shuffle_action_with_scatter ... ok
test configs::config_test::test_args_config ... ignored
test configs::config_test::test_config_file_not_found ... ignored
test clusters::cluster_test::test_add_node_with_clone ... ok
test configs::config_test::test_default_config ... ok
test configs::config_test::test_env_file_config ... ignored
test configs::config_test::test_file_config ... ignored
test clusters::node_test::test_serialize_node ... ok
test configs::config_test::test_fuse_commit_version ... ok
test api::rpc::flight_service_test::test_do_flight_action_with_shared_session ... ok
test datasources::common_test::test_util_generate_parts ... ok
test configs::config_test::test_env_config ... ok
test datasources::common_test::test_lines_count ... ok
test api::rpc::flight_service_test::test_do_flight_action_with_abort_session ... ok
test datasources::local::null_table_test::test_null_table ... ok
test datasources::local::memory_table_test::test_memorytable ... ok
test datasources::local::csv_table_test::test_csv_table ... ok
test datasources::system::clusters_table_test::test_clusters_table ... ok
test datasources::system::contributors_table_test::test_contributors_table ... ok
test datasources::system::databases_table_test::test_tables_table ... ok
test datasources::system::functions_table_test::test_functions_table ... ok
test datasources::local::parquet_table_test::test_parquet_table ... ok
test datasources::system::settings_table_test::test_settings_table ... ok
test datasources::system::numbers_table_test::test_number_table ... ok
test datasources::system::tables_table_test::test_tables_table ... ok
test datasources::system::tracing_table_test::test_tracing_table ... ok
test interpreters::interpreter_database_drop_test::test_drop_database_interpreter ... ok
test interpreters::interpreter_database_create_test::test_create_database_interpreter ... ok
test interpreters::interpreter_describe_table_test::interpreter_describe_table_test ... ok
test interpreters::interpreter_setting_test::test_setting_interpreter ... ok
test clusters::cluster_test::test_add_node_with_local ... ok
test interpreters::interpreter_show_create_table_test::interpreter_show_create_table_test ... ok
test interpreters::interpreter_table_create_test::test_create_table_interpreter ... ok
test interpreters::interpreter_table_drop_test::test_drop_table_interpreter ... ok
test interpreters::interpreter_truncate_table_test::test_truncate_table_interpreter ... ok
test api::rpc::flight_dispatcher_test::test_get_stream_with_non_exists_stream ... ok
test interpreters::interpreter_use_database_test::test_use_interpreter ... ok
test interpreters::plan_scheduler_test::test_scheduler_plan_with_convergent_and_expansive_stage ... ok
test interpreters::plan_scheduler_test::test_scheduler_plan_with_convergent_and_normal_stage ... ok
test interpreters::plan_scheduler_test::test_scheduler_plan_with_one_convergent_stage ... ok
test interpreters::plan_scheduler_test::test_scheduler_plan_without_stage ... ok
test metrics::metric_service_test::test_metrics_server ... ok
test functions::context_function_test::test_context_function_build_arg_from_ctx ... ok
test optimizers::optimizer_projection_push_down_test::test_projection_push_down_optimizer_1 ... ok
test optimizers::optimizer_projection_push_down_test::test_projection_push_down_optimizer_2 ... ok
test optimizers::optimizer_projection_push_down_test::test_projection_push_down_optimizer_3 ... ok
test interpreters::interpreter_setting_test::test_setting_interpreter_error ... ok
test datasources::local::csv_table_test::test_csv_table_parse_error ... ok
test optimizers::optimizer_statistics_exact_test::tests::test_statistics_exact_optimizer ... ok
test pipelines::processors::pipe_test::test_pipe ... ok
test api::rpc::flight_service_test::test_do_flight_action_with_abort_and_new_session ... ok
test interpreters::interpreter_use_database_test::test_use_database_interpreter_error ... ok
test optimizers::optimizer_projection_push_down_test::test_projection_push_down_optimizer_group_by ... ok
test pipelines::processors::processor_empty_test::test_processor_empty ... ok
test pipelines::processors::processor_merge_test::test_processor_merge ... ok
test pipelines::processors::pipeline_display_test::test_pipeline_display ... ok
test pipelines::processors::pipeline_walker_test::test_pipeline_walker ... ok
test datasources::tests::test_datasource ... ok
test pipelines::processors::processor_mixed_test::test_processor_mixed ... ok
test interpreters::interpreter_explain_test::test_explain_interpreter ... ok
test pipelines::transforms::transform_expression_test::test_transform_expression ... ok
test pipelines::processors::pipeline_builder_test::test_local_pipeline_builds ... ok
test pipelines::transforms::transform_filter_test::test_transform_filter ... ok
test pipelines::transforms::transform_group_by_final_test::test_transform_final_group_by ... ok
test pipelines::transforms::transform_group_by_partial_test::test_transform_partial_group_by ... ok
test pipelines::transforms::transform_limit_by_test::test_transform_limit_by ... ok
test pipelines::transforms::transform_limit_test::test_transform_limit ... ok
test pipelines::transforms::transform_projection_test::test_transform_projection ... ok
test pipelines::transforms::transform_sort_test::test_transform_sort ... ok
test pipelines::transforms::transform_source_test::transform_source_test ... ok
test pipelines::transforms::transform_aggregator_final_test::test_transform_final_aggregator ... ok
test pipelines::transforms::transform_aggregator_partial_test::test_transform_partial_aggregator ... ok
test pipelines::transforms::transform_expression_test::test_transform_expression_error ... ok
test servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_data ... ok
test servers::clickhouse::clickhouse_handler_test::test_abort_clickhouse_server ... ok
test servers::clickhouse::clickhouse_handler_test::test_clickhouse_handler_query ... ok
test servers::mysql::mysql_handler_test::test_rejected_session_with_parallel ... ok
test pipelines::transforms::transform_filter_test::test_transform_filter_error ... ok
test sql::sql_parser_test::tests::create_database ... ok
test sql::sql_parser_test::tests::describe_table ... ok
test sql::sql_parser_test::tests::drop_database ... ok
test sql::sql_parser_test::tests::drop_table ... ok
test sql::sql_parser_test::tests::hint_test ... ok
test sql::sql_parser_test::tests::show_queries ... ok
test sql::sql_parser_test::tests::truncate_table ... ok
test sql::sql_parser_test::tests::use_database_test ... ok
test sql::sql_parser_test::tests::create_table ... ok
test servers::mysql::mysql_handler_test::test_use_database_with_on_query ... ok
test sql::plan_parser_test::test_plan_parser ... ok
test optimizers::optimizer_scatters_test::test_scatter_optimizer ... ok
test optimizers::optimizer_constant_folding_test::tests::test_constant_folding_optimizer ... ok
test servers::clickhouse::clickhouse_handler_test::test_reject_clickhouse_connection ... ok
test servers::mysql::mysql_handler_test::test_rejected_session_with_sequence ... ok

test result: ok. 95 passed; 0 failed; 4 ignored; 0 measured; 0 filtered out; finished in 10.07s

     Running unittests (target/debug/deps/fuse_benchmark-065c51267fc370a3)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/fuse_query-71d863bd9e543965)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests (target/debug/deps/fuse_store-f28f54948d80645d)

running 79 tests
test api::rpc::flight_service_test::test_do_append has been running for over 60 seconds
test api::rpc::flight_service_test::test_flight_create_database has been running for over 60 seconds
test api::rpc::flight_service_test::test_flight_create_get_table has been running for over 60 seconds
test api::rpc::flight_service_test::test_flight_drop_table has been running for over 60 seconds
test api::rpc::flight_service_test::test_flight_generic_kv has been running for over 60 seconds
test api::rpc::flight_service_test::test_flight_get_database_meta_ddl_db has been running for over 60 seconds
test api::rpc::flight_service_test::test_flight_get_database_meta_ddl_table has been running for over 60 seconds
test api::rpc::flight_service_test::test_flight_get_database_meta_empty_db has been running for over 60 seconds
thread panicked while panicking. aborting.
error: test failed, to rerun pass '-p fuse-store --lib'

Caused by:
  process didn't exit successfully: `/Users/coswde/Source/fuse-query/target/debug/deps/fuse_store-f28f54948d80645d` (signal: 4, SIGILL: illegal instruction)
make: *** [unit-test] Error 101
```

**Comment by dantengsky** [comment]
have seen this before #1205, seems they are kind of "related"

**Comment by dantengsky** [comment]
git-bisect gives us the commit 04cde8d5ee9 , which introduces sled for the first time (in store)

the issue may be related to the way we construct the unit test.

A question issue is opened at sled: 

[https://github.com/spacejam/sled/issues/1356](https://github.com/spacejam/sled/issues/1356)

waiting for reply

**Comment by dantengsky** [comment]
BTW, for anyone wanting to investigate this issue:

As it is flaky, during the bisecting, please `cargo test` at least twice for each commit to ensuring the commit does not have the problem.

A single test run, especially the fresh build, tends to give a false-positive result. (instruction cache?)

**Comment by drmingdrmer** [comment]
fixed in #1326
